src/components/ProjectSearch/index.js

```diff
@@ -20,84 +23,256 @@ import {
+    const { results } = this.props;
+    return results
+      .toJS()
+      .filter(result => result.filepath && result.matches.length > 0);
+  };
+
+  getResultCount = () => {
+    const results = this.getResults();
+    return results.reduce(
+      (count, file) => count + (file.matches ? file.matches.length : 0),
+      0
+    );
+  };
+
+  onKeyDown = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Escape") {
+      return;
+    }
+
+    e.stopPropagation();
+
+    if (e.key !== "Enter") {
+      return;
+    }
+    this.focusedItem = null;
+    this.props.searchSources(this.state.inputValue);
+  };
+
+  onEnterPress = () => {
+    if (this.focusedItem && !this.state.inputFocused) {
+      const { setExpanded, file, expanded, match } = this.focusedItem;
+      if (setExpanded) {
+        setExpanded(file, !expanded);
+      } else if (match) {
+        this.selectMatchItem(match);
+      }
+    }
+  };
+
+  inputOnChange = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value;
+    this.setState({ inputValue });
+  };
+
+  renderFile(
+    file: Result,
+    focused: boolean,
+    expanded: boolean,
+    setExpanded: Function
+  ) {
+    if (focused) {
+      this.focusedItem = { setExpanded, file, expanded };
+    }
+
+    const matchesLength = file.matches.length;
+    const matches = ` (${matchesLength} match${matchesLength > 1 ? "es" : ""})`;
+
+    return (
+      <div
+        className={classnames("file-result", { focused })}
+        key={file.sourceId}
+        onClick={e => setExpanded(file, !expanded)}
+      >
+        <Svg name="arrow" className={classnames({ expanded })} />
+        <img className="file" />
+        <span className="file-path">{getRelativePath(file.filepath)}</span>
+        <span className="matches-summary">{matches}</span>
+      </div>
+    );
   }
 
-  isProjectSearchEnabled() {
-    return this.props.activeSearch === "project";
+  renderMatch(match: Match, focused: boolean) {
+    if (focused) {
+      this.focusedItem = { match };
+    }
+    return (
+      <div
+        className={classnames("result", { focused })}
+        onClick={() => setTimeout(() => this.selectMatchItem(match), 50)}
+      >
+        <span className="line-number" key={match.line}>
+          {match.line}
+        </span>
+        {this.renderMatchValue(match)}
+      </div>
+    );
   }
 
-  renderTextSearch() {
-    const {
-      sources,
-      results,
-      status,
-      searchSources,
-      closeProjectSearch,
-      selectLocation,
-      textSearchQuery
-    } = this.props;
+  renderMatchValue(lineMatch: Match) {
+    return highlightMatches(lineMatch);
+  }
+
+  renderResults() {
+    const results = this.getResults().filter(
+      result => result.matches.length > 0
+    );
+
+    const { status } = this.props;
+
+    function getFilePath(item: Match, index?: number) {
+      return item.filepath
+        ? `${item.sourceId}-${index || "$"}`
```

Comment:
why the $?

Reply:
`ManagedTree` doesn't ever directly pass the index back but `Tree` which `ManagedTree` wraps does, but the effect of this is that you sometimes do and don't have an index so to avoid stringifying `undefined` I put the `$`. This speaks to issues in `ManagedTree` and maybe `Tree` or just the relationship between the two but that should be addressed on its own imo.